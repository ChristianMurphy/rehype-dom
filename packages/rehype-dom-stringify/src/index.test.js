import unified from 'unified';

import parse from './index';

describe('rehype-dom-stringify', () => {
  const processor = unified().use(parse).freeze();

  it('should stringify a complete document', () => {
    const result = processor()
      .data('settings', { fragment: false })
      .stringify({
        type: 'root',
        children: [{
          type: 'element',
          tagName: 'html',
          properties: {},
          children: [{
            type: 'element',
            tagName: 'head',
            properties: {},
            children: [{
              type: 'element',
              tagName: 'title',
              properties: {},
              children: [{
                type: 'text',
                value: 'Hi',
              }],
            }],
          }, {
            type: 'element',
            tagName: 'body',
            properties: {},
            children: [{
              type: 'element',
              tagName: 'h2',
              properties: {},
              children: [{
                type: 'text',
                value: 'Hello world!',
              }],
            }],
          }],
        }],
      });
    expect(result).toMatchSnapshot();
  });

  it('should stringify a fragment', () => {
    const result = processor()
      .data('settings', { fragment: true })
      .stringify({
        type: 'root',
        children: [{
          type: 'element',
          tagName: 'title',
          properties: {},
          children: [{
            type: 'text',
            value: 'Hi',
          }],
        }, {
          type: 'element',
          tagName: 'h2',
          properties: {},
          children: [{
            type: 'text',
            value: 'Hello world!',
          }],
        }],
      });
    expect(result).toMatchSnapshot();
  });
});
