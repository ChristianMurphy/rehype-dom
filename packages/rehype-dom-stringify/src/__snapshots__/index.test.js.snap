// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rehype-dom-stringify should stringify a complete document 1`] = `"<html><head><title>Hi</title></head><body><h2>Hello world!</h2></body></html>"`;

exports[`rehype-dom-stringify should stringify a fragment 1`] = `"<title>Hi</title><h2>Hello world!</h2>"`;
